### DISCUSSION TITLE
Feature request: Create a new branch from a tag in GitHub

### DISCUSSION THREAD

**Discussion by indranilchak** [community/community#138283] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Is it possible to create a new branch from a tag in GitHub? I only see the option to create a branch from other branches, not from a specific commit or a tag. I know the command line route but I was wondering if it's possible to do that from the web interface.

[![New Branch][1]][1]

  [1]: https://i.sstatic.net/pVxZQNfg.png

**Comment by spenserblack** [comment, marked as answer]
First you have to be browsing that specific tag or commit. Then, on the repo homepage, from the branch/tag dropdown, start typing in the name of your new branch. *You must be on the branch tab before you start typing.* If you start typing from the tag tab and then switch to the branch tab, you won't get the prompt to create a new branch.

Edit: As https://github.com/orgs/community/discussions/138283#discussioncomment-15424781 pointed out this no longer works.

**Reply by ebndev** [reply]
@spenserblack thanks for taking the time to answer! 😄

**Reply by JaixTim** [reply]
> First you have to be browsing that specific tag or commit. Then, ...

That's not at all intuitive. I've just done it based on this answer but it would be a lot simpler to be able to do it from the general create branch dialog.

**Reply by zeetaros** [reply]
I think GitHub realized this workaround was a bug and now patched it. After typing new branch name on the branch tab, it indeed shows "create new branch .... " option, but as soon as you switch to tag tab, the option is gone. At the end you still cannot create branch from a tag on the web version.

**Reply by spenserblack** [reply]
> but as soon as you switch to tag tab, the option is gone. At the end you still cannot create branch from a tag on the web version.

That's unfortunate. Besides the original behavior allowing you to branch from a tag, you could also branch from any arbitrary commit IIRC if you were browsing at that commit. That was really useful for going back a few steps and trying a different implementation on a new branch.

**Reply by zeetaros** [reply]
Guess using the terminal is the only way ( :

**Comment by mtg-github-full-admin** [comment]
I know this is old, but just had the same question. Is there any movement to revert/fix/supply this?

Is the only other option using a github action that can be run from the repo (would have to get added to every repo, but that could get automated)?

**Comment by Mealman1551** [comment]
this is possible:

1. navigate to the tag page
2. click under the tag name on the commit link
3. click "Browse files"
4. in the commit repository tree, make a new branch

~_Mealman1551_

**Reply by bbober-gd** [reply]
@Mealman1551  or anyone else: Please provide screenshots. I navigated from the tag to the commit to the source at that commit version and don't see an option to create a branch. I had to find the tag locally and do it on the command-line.
<img width="1188" height="103" alt="image" src="https://github.com/user-attachments/assets/7f5f8b0d-5971-468b-ad82-a2bf281211d1" />

Btw, the fact the original method from 2024 no longer works explains why this must be a first-class feature to avoid regressions in this workflow.

**Reply by robhogan** [reply]
To elaborate on 4 - start typing in the dropdown showing the commit you want to branch from, then the option will be "create _branch_ from _commit_"

<img width="362" height="309" alt="image" src="https://github.com/user-attachments/assets/641bc883-0d92-4158-9722-9d9dd3c4f161" />
